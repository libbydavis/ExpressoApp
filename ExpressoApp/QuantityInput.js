import React, {Component} from "react";
import { Image, TextInput, TouchableOpacity, View, StyleSheet, Alert } from "react-native";


export default class quantityInput extends Component {
    state = {
        quantityText: '5',
        quantityValue: 5,
        quantityPlaceholder: '5',
    }
    handleButtonQuantity(buttonType) {
        let currentValue = this.state.quantityValue;
        if (buttonType == true) {
            currentValue = ++this.state.quantityValue;
        }
        else if (currentValue - 1 < 0) {
            Alert.alert("You can't have a value smaller than 0!");
        }
        else if (currentValue - 1 >= 0) {
            currentValue = --this.state.quantityValue;
        }
        let textValue = currentValue.toString();

        this.setState({
            quantityValue: currentValue,
            quantityText: textValue,
        });
    }

    minusQuantity() {
        this.handleButtonQuantity(false);
    }

    plusQuantity() {
        this.handleButtonQuantity(true);
    }

    setKeyQuantity() {
        let textValue = "10";
        let currentValue = textValue.valueOf();

        this.setState({
            quantityValue: currentValue,
            quantityText: textValue,
        });
        console.log(this.state.quantityText);
    }

    render() {
        return(
            <View style={styles.quantityRow}>
                <TouchableOpacity onPress={this.minusQuantity.bind(this)} style={styles.touchableIcons}>
                    <Image source={require('./assets/minusButton.png')} style={styles.quantityIcon} />
                </TouchableOpacity>
                <TextInput style={styles.quantityInputBox} placeholder={this.state.quantityPlaceholder} value={this.state.quantityText} onKeyPress={this.setKeyQuantity.bind(this)} />
                <TouchableOpacity onPress={this.plusQuantity.bind(this)} style={styles.touchableIcons}>
                    <Image source={require('./assets/addButton.png')} style={styles.quantityIcon} />
                </TouchableOpacity>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    quantityRow: {
        flexDirection: "row",
    },
    quantityIcon: {
        width: 25,
        height: 25,
    },
    quantityInputBox: {
        marginLeft: 10,
    },
    touchableIcons: {
        marginTop: 10,
    },
});
