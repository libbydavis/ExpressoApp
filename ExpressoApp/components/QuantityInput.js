import React, {useEffect, useState} from 'react';
import {
  Image,
  TextInput,
  TouchableOpacity,
  View,
  StyleSheet,
  Alert,
} from 'react-native';

function setInitial(props) {
  if (props != undefined) {
    let quan = props.initialQuantity ? props.initialQuantity : 1;
    return quan;
  }
  return 1;
}

const QuantityInput = (props) => {
  const [quantity, setQuantity] = useState(setInitial(props));


  const handleButtonQuantity = (buttonType) => {
    let newQuantity;
    if (buttonType == true) {
      newQuantity = quantity + 1;
      setQuantity(newQuantity);
    } else if (quantity - 1 < 0) {
      Alert.alert('You can\'t have a value smaller than 0!');
    } else if (quantity - 1 >= 0) {
      newQuantity = quantity - 1;
      setQuantity(newQuantity);
    }
    props.receiveValue(newQuantity);
  };

  const handlePlus = () => {
    handleButtonQuantity(true);
  };

  const handleMinus = () => {
    handleButtonQuantity(false);
  };

  const handleTextQuantity = (text) => {
    setQuantity(Number(text));
    props.receiveValue(Number(text));
  };

  return (
    <View style={styles.quantityRow}>
      <TouchableOpacity onPress={() => handleMinus()}>
        <Image
          source={require('../assets/minusButton.png')}
          style={styles.quantityIcon}
        />
      </TouchableOpacity>
      <TextInput
        style={styles.quantityInputBox} value={quantity.toString()}
        onChangeText={(text) => handleTextQuantity(text)}
        keyboardType={'numeric'}
      />
      <TouchableOpacity onPress={() => handlePlus()}>
        <Image
          source={require('../assets/addButton.png')}
          style={styles.quantityIcon}
        />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  quantityRow: {
    flexDirection: 'row',
  },
  quantityIcon: {
    width: 25,
    height: 25,
  },
  quantityInputBox: {
    marginLeft: 10,
    textAlignVertical: 'top',
  },
});

export default QuantityInput;
