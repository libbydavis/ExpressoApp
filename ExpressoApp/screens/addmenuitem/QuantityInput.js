import React, {useState} from 'react';
import {
  Image,
  TextInput,
  TouchableOpacity,
  View,
  StyleSheet,
  Alert,
} from 'react-native';


const QuantityInput = (props) => {
  const setInitialQuantity = (initial) => {
    return initial? initial : 0;
  }

  const [quantity, setQuantityNumber] = useState(() => setInitialQuantity(props.initialQuantity));

  const handleButtonQuantity = (buttonType) => {
    let newQuantity;
    if (buttonType == true) {
      newQuantity = quantity + 1;
      setQuantityNumber(newQuantity);
    } else if (quantity - 1 < 0) {
      Alert.alert('You can\'t have a value smaller than 0!');
    } else if (quantity - 1 >= 0) {
      newQuantity = quantity - 1;
      setQuantityNumber(newQuantity);
    }
    props.receiveValue(newQuantity);
  };

  const handlePlus = () => {
    handleButtonQuantity(true);
  };

  const handleMinus = () => {
    handleButtonQuantity(false);
  };

  const handleTextQuantity = (text) => {
    setQuantityNumber(Number(text));
    props.receiveValue(Number(text));
  };

  return (
      <View style={styles.quantityRow}>
        <TouchableOpacity onPress={() => handleMinus()}>
          <Image
              source={require('../../assets/minusButton.png')}
              style={styles.quantityIcon}
          />
        </TouchableOpacity>
        <TextInput
            style={styles.quantityInputBox} value={quantity.toString()}
            onChangeText={(text) => handleTextQuantity(text)}
            keyboardType={'numeric'}
        />
        <TouchableOpacity onPress={() => handlePlus()}>
          <Image
              source={require('../../assets/addButton.png')}
              style={styles.quantityIcon}
          />
        </TouchableOpacity>
      </View>
  );
};

const styles = StyleSheet.create({
  quantityRow: {
    flexDirection: 'row',
    marginBottom: 20,
  },
  quantityIcon: {
    width: 25,
    height: 25,
  },
  quantityInputBox: {
    marginLeft: 10,
    textAlignVertical: 'top',
  },
});

export default QuantityInput;
