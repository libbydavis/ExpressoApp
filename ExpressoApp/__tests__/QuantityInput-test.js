// __tests__/QuantityInput-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import QuantityInput from '../components/QuantityInput';
import handlePlus from '../components/QuantityInput';
import handleMinus from '../components/QuantityInput';
import quantity from '../components/QuantityInput';
import {shallow} from "react-native/jest/renderer";


test('renders correctly', () => {
    const quanInput = renderer.create(<QuantityInput initialQuantity={5}/>).toJSON();
    expect(quanInput).toMatchSnapshot();
});


describe('quantity add and minus', () => {
    it('test add and minus', () => {
        handlePlus()
        expect(quantity.toBe(2));
        handleMinus()
        expect(quantity.toBe(1));
    })
})


