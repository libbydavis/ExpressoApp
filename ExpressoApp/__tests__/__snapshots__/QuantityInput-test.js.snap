// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/minusButton.png",
        }
      }
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "marginLeft": 10,
        "textAlignVertical": "top",
      }
    }
    underlineColorAndroid="transparent"
    value="5"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/addButton.png",
        }
      }
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
  </View>
</View>
`;
